### ISSUE TITLE
Event type for error event is incorrect

### ISSUE THREAD

**Issue by upsuper** [microsoft/TypeScript-DOM-lib-generator#1377]
In `GlobalEventHandlersEventMap`, `error` event is mapped to `ErrorEvent`. This might be correct for `Window` when handling events from runtime error, it is incorrect for `Element`s. When `Element`s trigger that event, it dispatches either a `UIEvent` or a `Event`, but never an `ErrorEvent`, see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/error_event).

If you put the following code into browser console, you can see this behavior clearly:
```javascript
const img = new Image();
img.addEventListener('error', e => console.log(e));
img.src = 'invalid';
```

So mapping `error` event to `ErrorEvent` for elements is misleading.

**Comment by jpc-ae** [comment]
Note that it's also potentially incorrect for the Window element if the error was not generated from a user interface element.
Ref: https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event#event_type

Since an `error` can always result in a basic `Error` event, it should be mapped as such. `UIEvent` or `ErrorEvent` can then be tested for in the callback to confirm the origin of an error if needed.

It looks like what happened here is the ErrorEvent mapping was put in the wrong place. According to MDN, it should be the standard value for the first argument of `onerror` (`OnErrorEventHandlerNonNull`).
Ref: https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event#error

While this can potentially be a breaking change for anyone using the properties of ErrorEvent inside an event handler without checking what type of event was fired, the current behaviour is misleading at best and should be changed to alert authors of the potential type errors hidden here.

To summarize, the following basic changes should be made to align with the standards:

`Event` instead of `ErrorEvent` on line 16691:
```ts
interface GlobalEventHandlersEventMap {
    // ...
    "error": Event;
    // ...
}
```

`ErrorEvent` instead of `Event` on line 42996:
```ts
interface OnErrorEventHandlerNonNull {
    (event: ErrorEvent | string, source?: string, lineno?: number, colno?: number, error?: Error): any;
}
```
